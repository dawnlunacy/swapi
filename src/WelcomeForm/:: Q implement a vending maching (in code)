// Q implement a vending maching (in code)

- What are the available items and their prices
- How many items are in inventory
- Buy
- Restock

vm = VendingMachine.new
vm.inventory => "Chocolate: $3, Candy $4"

vm.inventory("Chocolate")=>5

vm.buy("Chocolate", 3)
vm.inventory("Chocolate")=> 2

vm.restock("Chocolate", 10)
vm.inventory("Chocolate") =>  12

// class of vending machine
//takes in an array of inventory
  


  [{
    candy: { inventory: #, pricePerOne: $}}, {}, {}]

  // how many items are in inventory = reduce over array and add up the inventory

  // to buy: // find key === candy // check inventory !== 0 , // return price  

    // method that takes in id for candy


  //



  //// consuelo

  this.snack = [{name:chocolate, price: #, quantity: #}]

  buySnacks(snack)

  this.snacks.find(snack => snack.name === snack)



**** Be sure to psuedocode -- don't write in actual code, talking through it will show that you know the syntax

**** Negotiation panel **** next week
*** devOps and IOT engineering *** 

/// she will share the problems // work out in gist and share 






