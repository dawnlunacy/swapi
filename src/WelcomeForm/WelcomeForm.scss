h1 {
    text-align:center;
    position: relative;
    bottom: 42px;
}

.welcome-form-container {
    background-color: cornsilk;
    height: 700px;
    width: 800px;
    // display: flex;
    // justify-content: center;
    // align-items: center;
    margin-top: 50px;
}

form {
    // display: block;
    background-color: plum;
    width: 600px;
    height:500px;
    display: flex;
    flex-direction: column;
    /* justify-content: center; */
    align-items: center;
    margin-left: 4.5em;
    padding: 30px;
}

input, label {
    display: block;
    color: black;
    text-align: center;

}

input {
    font-size: 2.5em;
    padding-left: 6px;
    width: 600px;
    margin-bottom: 5px;
}

label {
    font-size: 1.5em;
    margin-bottom: 2px;
}

select {
    font-size: 2.5em;
    width: 600px;
    text-align: center;
}

button {
    font-size: 1.5em;
    margin-top: 4em;
}

.error {
    color: red;
    margin-bottom: 20px;
}