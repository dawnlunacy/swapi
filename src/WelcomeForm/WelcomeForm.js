import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './WelcomeForm.scss';
import {  Link } from 'react-router-dom';
import App from '../App'



class WelcomeForm extends Component {
    constructor() {
        super()
        this.state = {
            name: '',
            quote: '',
            level:'select',
            formReady: false,
            buttonText: 'Submit I Shall Not'
        }
    }

    validate = ({ name, quote, level }) => {

        return {
          name:
            !name || name.trim().length === 0
              ? "* Name is required *"
              : false,
          quote:
            !quote || quote.trim().length === 0
              ? "* Quote is required *"
              : false,
          level:
            !level || level === 'select'
              ? "* Rank is required *"
              : false
        //  formReady:
            
        };
    };

    displayUserInfo = () => {
        const { userInfo } = this.props
        userInfo(this.state.name, this.state.quote, this.state.level)
    }
    
    handleChange = event => {
        this.setState({[event.target.name]: event.target.value})

        if ( this.state.name !== ''  && this.state.quote !== '' && this.state.level !== 'select' ) {
            this.setState({buttonText:"Submit I Shall"});
            this.setState({formReady: true}) 
        }
    }

    render() {
        const errors = this.validate(this.state)
    
        return (
            <div className="welcome-form-container">
                <form>
                    <h1>Welcome Jedi</h1>
                    <label>Name:</label>
                    <input
                    type="text"
                    placeholder="Enter Name"
                    name="name"
                    value={this.state.name}
                    onChange= { this.handleChange }
                    />
                    {errors && <span className="error"> { errors.name } </span>}

                    <label>Favorite Quote:</label>
                    <input
                     type="text"
                     placeholder="Enter Quote"
                     name="quote"
                     value={this.state.quote}
                     onChange= { this.handleChange }
                     />
                    {errors && <span className="error"> { errors.quote } </span>}

                    <label>Select Rank:</label>
                    <select onChange={this.handleChange.bind(this)} name="level" value={this.state.level}>
                        <option  value="select">Select a Rank</option>
                        <option  value="Padawan">Padawan</option>
                        <option  value="Knight">Knight</option>
                        <option  value="Master">Master</option>
                    </select>
                    {errors && <span className="error"> { errors.level } </span>}
                    <Link to='/movies' className='nav'> <button onClick={this.displayUserInfo} disabled={!this.state.formReady}> {this.state.buttonText}</button> </Link>
                </form>
            </div>
        )
    }
}

export default WelcomeForm;

WelcomeForm.propTypes = {
    movies: PropTypes.array.isRequired,
  }