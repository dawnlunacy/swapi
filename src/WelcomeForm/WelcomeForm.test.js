import React from 'react';
import { shallow } from 'enzyme';
import WelcomeForm from '../WelcomeForm/WelcomeForm';

describe('WelcomeForm', () => {
    let wrapper
    let state = {
        movies:[],
        currentCharacters:[],
        userData: '',
        isLoading: true,
        name: '',
        quote: '',
        level: ''
    }
    const mockGetMovieCharacters = jest.fn()
    const mockUserInfo = jest.fn()
    beforeEach(() => {
        wrapper = shallow(<WelcomeForm {...state} getMovieCharacters={mockGetMovieCharacters} userInfo={mockUserInfo}/>)
    })
    it('should match the snapshot with all the correct data passed through', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should update the state of name when handleChange is being called', () => {
    
        const mockEvent = {target: {name: 'name', value: 'Sausage'}}

        const expected = 'Sausage'


        wrapper.instance().handleChange(mockEvent);

        expect(wrapper.state('name')).toEqual(expected);
    });
    it('should update the state of quote when handleChange is being called', () => {
    
        const mockEvent = {target: {name: 'quote', value: 'I am Spicy'}}

        const expected = 'I am Spicy'


        wrapper.instance().handleChange(mockEvent);

        expect(wrapper.state('quote')).toEqual(expected);
    });
    it('should update the state of quote when handleChange is being called', () => {
    
        const mockEvent = {target: {name: 'level', value: 'Master'}}

        const expected = 'Master'


        wrapper.instance().handleChange(mockEvent);

        expect(wrapper.state('level')).toEqual(expected);
    });
    it('should run displayUserInfo when the button is clicked', () => {
        wrapper.instance().displayUserInfo = jest.fn();
        wrapper.find('button').simulate('click')

        expect(mockUserInfo).toHaveBeenCalled()
    })
})