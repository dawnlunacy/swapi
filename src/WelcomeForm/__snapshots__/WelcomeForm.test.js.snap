// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeForm should match the snapshot with all the correct data passed through 1`] = `
<div
  className="welcome-form-container"
>
  <form>
    <h1>
      Welcome Jedi
    </h1>
    <label>
      Name:
    </label>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Enter Name"
      type="text"
      value=""
    />
    <span
      className="error"
    >
       
      * Name is required *
       
    </span>
    <label>
      Favorite Quote:
    </label>
    <input
      name="quote"
      onChange={[Function]}
      placeholder="Enter Quote"
      type="text"
      value=""
    />
    <span
      className="error"
    >
       
      * Quote is required *
       
    </span>
    <label>
      Select Rank:
    </label>
    <select
      name="level"
      onChange={[Function]}
      value="select"
    >
      <option
        value="select"
      >
        Select a Rank
      </option>
      <option
        value="Padawan"
      >
        Padawan
      </option>
      <option
        value="Knight"
      >
        Knight
      </option>
      <option
        value="Master"
      >
        Master
      </option>
    </select>
    <span
      className="error"
    >
       
      * Rank is required *
       
    </span>
    <Link
      className="nav"
      to="/movies"
    >
       
      <button
        disabled={true}
        onClick={[Function]}
      >
         
        Submit I Shall Not
      </button>
       
    </Link>
  </form>
</div>
`;
